### PR TITLE
Double indent wrapped lines.

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: src/main/java/com/squareup/javapoet/TypeSpec.java [hunk 1]
```diff
@@ -103,126 +103,136 @@ public static Builder anonymousClassBuilder(String typeArgumentsFormat, Object..
...
-      codeWriter.emit("$L", enumName);
-      if (!anonymousTypeArguments.formatParts.isEmpty()) {
-        codeWriter.emit("(");
+    // Nested classes interrupt wrapped line indentation. Stash the current wrapping state and put
+    // it back afterwards when this type is complete.
+    int previousStatementLine = codeWriter.statementLine;
+    codeWriter.statementLine = -1;
+
+    try {
+      if (enumName != null) {
```
Comment (swankjesse) [inline src/main/java/com/squareup/javapoet/TypeSpec.java]: FYI, everything in the `try { }` is unchanged; just indented + 2.

**Comment by swankjesse** [comment]
I'm excited about this. The `@Override` line is indented 6 total when the type its declared inside is on the first line of a statement, and indented 10 total when that type is on a subsequent line. (This is consistent with IntelliJ, but somewhat counter-intuitive.)

**Comment by swankjesse** [comment]
@mattprecious, this one's for you

File: src/test/java/com/squareup/javapoet/TypeSpecTest.java [hunk 2]
```diff
@@ -1099,4 +1101,93 @@
...
+        + "    return \"Taco(\"\n"
+        + "        + \"beef,\"\n"
+        + "        + \"lettuce,\"\n"
+        + "        + \"cheese\"\n"
+        + "        + \")\";\n"
+        + "  }\n"
+        + "}\n");
+  }
+
+  @Test public void multilineStatementWithAnonymousClass() throws Exception {
```
Comment (JakeWharton) [inline src/test/java/com/squareup/javapoet/TypeSpecTest.java]: Nice test.

**Pull request merged**